### PR TITLE
line chartV9

### PR DESCRIPTION
**Pull request by Anush2303**
## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { tokens } from '@fluentui/react-theme';
+import { ILineChartProps, ILineChartStyles } from './LineChart.types';
+import { SlotClassNames } from '@fluentui/react-utilities/src/index';
+
+/**
+ * @internal
+ */
+export const linechartClassNames: SlotClassNames<ILineChartStyles> = {
+  tooltip: 'fui-lc__tooltip',
+  root: '',
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts:11]: fill these values

File: packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
+import { tokens } from '@fluentui/react-theme';
+import { ILineChartProps, ILineChartStyles } from './LineChart.types';
+import { SlotClassNames } from '@fluentui/react-utilities/src/index';
+
+/**
+ * @internal
+ */
+export const linechartClassNames: SlotClassNames<ILineChartStyles> = {
+  tooltip: 'fui-lc__tooltip',
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts:10]: call this fui-line.....

File: packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ */
+const useStyles = makeStyles({
+  tooltip: {
+    display: 'flex',
+    flexDirection: 'column',
+    ...shorthands.padding('8px'),
+    position: 'absolute',
+    textAlign: 'center',
+    top: '0px',
+    fill: tokens.colorNeutralBackground1,
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts:42]: I see background property is missing

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx [hunk 4]
```diff
@@ -1,16 +1,1312 @@
+/* eslint-disable @typescript-eslint/no-shadow */
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:1]: How can these 2 bypasses be removed?

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx [hunk 5]
```diff
@@ -1,16 +1,1312 @@
...
-import { LineChartBase } from './LineChart.base';
-import { getStyles } from './LineChart.styles';
+import { ILineChartProps } from './LineChart.types';
+import { useLineChartStyles_unstable } from './useLineChartStyles.styles';
+import { Axis as D3Axis } from 'd3-axis';
+import { select as d3Select, pointer } from 'd3-selection';
+import { bisector } from 'd3-array';
+import { ILegend, Legends } from '../Legends/index';
+import { line as d3Line, curveLinear as d3curveLinear } from 'd3-shape';
+import { getId, find, memoizeFunction } from '@fluentui/react/lib/Utilities';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:11]: All references from react/lib/utilities need to be removed

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx [hunk 6]
```diff
@@ -1,16 +1,1312 @@
...
+  IChildProps,
+  ILineChartPoints,
+  ICustomizedCalloutData,
+  IMargins,
+  IRefArrayData,
+  IColorFillBarsProps,
+  ILineChartGap,
+  ILineChartDataPoint,
+} from '../../index';
+import { DirectionalHint } from '@fluentui/react/lib/Callout';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:25]: This dependency needs to be removed

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx [hunk 7]
```diff
@@ -1,16 +1,1312 @@
...
+    const calloutPointsRef = React.useRef([]);
+    React.useEffect(() => {
+      /** note that height and width are not used to resize or set as dimesions of the chart,
+       * fitParentContainer is responisble for setting the height and width or resizing of the svg/chart
+       */
+      if (_points !== _injectIndexPropertyInLineChartData(props.data.lineChartData) || props.data !== _points) {
+        pointsRef.current = _injectIndexPropertyInLineChartData(props.data.lineChartData);
+        calloutPointsRef.current = calloutData(pointsRef.current);
+      }
+    });
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:184]: add a effect trigger condition of change in height, weight and data

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx [hunk 8]
```diff
@@ -1,16 +1,1312 @@
...
+      const point = line.data[pointIndex];
+      const formattedDate = point.x instanceof Date ? point.x.toLocaleString() : point.x;
+      const xValue = point.xAxisCalloutData || formattedDate;
+      const legend = line.legend;
+      const yValue = point.yAxisCalloutData || point.y;
+      return point.callOutAccessibilityData?.ariaLabel || `${xValue}. ${legend}, ${yValue}.`;
+    }
+
+    function _isChartEmpty(): boolean {
+      /*return !(
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:1199]: why comment out this code

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx [hunk 9]
```diff
@@ -1,16 +1,1312 @@
...
-import { styled } from '@fluentui/react/lib/Utilities';
-import { ILineChartProps, ILineChartStyleProps, ILineChartStyles } from './LineChart.types';
-import { LineChartBase } from './LineChart.base';
-import { getStyles } from './LineChart.styles';
+import { ILineChartProps } from './LineChart.types';
+import { useLineChartStyles_unstable } from './useLineChartStyles.styles';
+import { Axis as D3Axis } from 'd3-axis';
+import { select as d3Select, pointer } from 'd3-selection';
+import { bisector } from 'd3-array';
+import { ILegend, Legends } from '../Legends/index';
```
[the remark this answers is not in the document]
Comment (Anush2303) [reply packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:9]: Okay, I have started with VBC for now. After that, should I go with legends?

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx (hunk 7 above)
Comment (Anush2303) [reply packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:184]: Okay

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx (hunk 8 above)
Comment (Anush2303) [reply packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:1199]: By mistake, I was using this to debug the code. Will change this.

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (AtishayMsft) [reply packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:9]: Added the plan in onenote. Will discuss tomorrow.

File: packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts (hunk 3 above)
Comment (Anush2303) [reply packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts:42]: Okay, will add it.

Comment (Anush2303) [reply packages/react-components/react-charts-preview/src/components/LineChart/useLineChartStyles.styles.ts:42]: Won't the fill property do work for background ?

**Comment by Anush2303** [comment]
Committed with required changes.

**Review by AtishayMsft — APPROVED** [review]

**Comment by AtishayMsft** [comment]
Approved but there is a version mismatch error reported by the build pipeline.

**Pull request merged**